src/org/daisy/dotify/formatter/impl/obfl/ObflParserImpl.java

```diff
@@ -1603,11 +1591,27 @@ private void parseExternalReference(
         Map<QName, String> attrList = new HashMap<>();
         while (it.hasNext()) {
             Attribute a = it.next();
+            MetaDataItem newItem = new MetaDataItem(
+                new QName(null, a.getName().getPrefix(), "xmlns"),
+                a.getName().getNamespaceURI()
+            );
+            if (!checkIfAlreadyContainsPrefix(meta, newItem)) {
+                meta.add(newItem);
+            }
             attrList.put(a.getName(), a.getValue());
         }
         fc.insertExternalReference(attrList);
     }
 
+    private boolean checkIfAlreadyContainsPrefix(List<MetaDataItem> meta, MetaDataItem newItem) {
```

Comment:
Actually it's better to check for the namespace URIs, not the prefixes. It is OK to have two mappings from the same prefix to different URIs, or two mappings from different prefixes to the same URI. The PEF writer can cope with that. But when you don't create a metadata item for a namespace that is used in the OBFL (because used prefix already mapped to a different URI), the writer can also not declare it on the `pef` element (with a generated prefix). Then it will have to declare it on the `row` element.

Reply:
Hi @bertfrees

Sound like a reasonable change. Updated to use namespaceURI instead.

Best regards
Daniel